### PR TITLE
Optics filterindex typeclass

### PR DESCRIPTION
**Pull request by nomisRev**

### CODE REVIEW

File: arrow-optics/src/main/kotlin/arrow/optics/std/option.kt [hunk 1]
```diff
@@ -40,9 +39,31 @@ fun <A> nonePrism(): Prism<Option<A>, Unit> = Prism(
...
+ */
+fun <A> optionOptional(): Optional<Option<A>, A> = Optional(
+        getOrModify = { a -> a.fold({ a.left() }, { it.right() }) },
+        set = { a -> { it.fold({ Option.empty() }, { a.toOption() }) } }
+)
+
+/**
+ * [Optional] to safely operate on a nullable value.
+ */
+fun <A> nullableOptional(): Optional<A?, A> = Optional(
```
Comment (raulraja) [inline arrow-optics/src/main/kotlin/arrow/optics/std/option.kt]: :clap:

**Comment by nomisRev** [comment]
Should I add the docs under `typeclasses` or `optics`?

**Comment by raulraja** [comment]
Since it's optics related I think optics is a better place

**Comment by nomisRev** [comment]
@raulraja Updated with docs.

File: arrow-docs/docs/docs/optics/filterindex/README.MD [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Given a `FilterIndex` instance we can create a `Traversal` that filters out the foci that do not match the predicate.
+
+```kotlin:ank
+val filter: Traversal<ListKWKind<String>, String> = filterIndexStringByIndex.filter { length -> length > 3 }
+
+filter.getAll(listOf("H", "He", "Hel", "Hell", "Hello").k())
+```
+
+Arrow provides some instances like `ListKW` indexed by index position.
```
Comment (pakoito) [inline arrow-docs/docs/docs/optics/filterindex/README.MD]: I'm not sure what this means, and I'd reference the call required to get these instances in this paragraph so people understand which part of the snippet you're referring to.

**Pull request merged**